x/twap/logic_test.go

```diff
@@ -791,6 +780,51 @@ func TestComputeArithmeticTwap_ThreeAsset(t *testing.T) {
 	}
 }
 
+func TestComputeArithmeticTwap_ThreeAsset_Geometric(t *testing.T) {
+	var (
+		five        = sdk.NewDec(5)
+		fiveFor3Sec = OneSec.MulInt64(3).Mul(twap.TwapLog(five))
+
+		ten          = five.MulInt64(2)
+		tenFor100Sec = OneSec.MulInt64(100).Mul(twap.TwapLog(ten))
+
+		errTolerance = sdk.MustNewDecFromStr("0.00000001")
```

Comment:
How is this chosen?

Reply:
Can we make an issue to revisit this error tolerance in the lens of stableswap with high scaling factors?

Reply:
This was chosen as the tightest allowed precision with the current `Pow` function. Should be possible to make tighter once the new `Power` function is merged.

Will make an issue

Reply:
https://github.com/osmosis-labs/osmosis/issues/3770